### PR TITLE
#133 Adding id to custom claims on sign up and sign in

### PR DESCRIPTION
**Pull request by Ymkavi**
Environment variable "Firebase_Service_Account_Private_Key" with private key must be added

### CODE REVIEW

**Review by apofis96 — CHANGES_REQUESTED** [review]
Fix conflicts please.
And git warnings

**Comment by apofis96** [comment]
Add an method to obtain identity from context.

**Review by apofis96 — APPROVED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/UserService.cs [hunk 1]
```diff
@@ -62,7 +74,46 @@ public async Task<UserDto> CreateUserPreferences(NewUserDto userDto)
...
+            };
+
+            await _firebaseAuth.SetCustomUserClaimsAsync(uid, userClaims);
+        }
+
+        public string GetCurrentUserEmail()
+        {
+            var claimsList = _httpContextAccessor.HttpContext!.User.Claims.ToList();
+            var email = claimsList.Find(el => el.Type == ClaimTypes.Email);
+            return email!.Value;
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/UserService.cs]: Why not Id?
And I believe we already have this method

File: backend/EasyMeets.Core/EasyMeets.Core.WebAPI/service-account-file.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  "type": "service_account",
+  "project_id": "easymeets-4ed60",
+  "private_key_id": "5c4f4c464f15e8e4ec2155870590b7faedb57d90",
+  "client_email": "firebase-adminsdk-ud2yz@easymeets-4ed60.iam.gserviceaccount.com",
+  "client_id": "111694983450380510761",
+  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+  "token_uri": "https://oauth2.googleapis.com/token",
+  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/firebase-adminsdk-ud2yz%40easymeets-4ed60.iam.gserviceaccount.com"
+}
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.WebAPI/service-account-file.json]: Please move private items to environment variables

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.WebAPI/service-account-file.json (hunk 2 above)
Comment (tatianahutii) [reply backend/EasyMeets.Core/EasyMeets.Core.WebAPI/service-account-file.json]: And delete this file
Not security to appsetting file
Secure to environment variables

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/UserService.cs [hunk 3]
```diff
@@ -92,6 +143,13 @@ public async Task<ImagePathDto> UploadImageAsync(IFormFile file)
             return new ImagePathDto() { Path = imagePath };
         }
 
+        public async Task<long> GetIdFromClaims(string uid)
+        {
+            var userRecord = await _firebaseAuth.GetUserAsync(uid);
```
Comment (apofis96) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/UserService.cs]: Why you don't use claims from context?
Move this method to appropriate place.

**Review by apofis96 — CHANGES_REQUESTED** [review]

**Review by apofis96 — APPROVED** [review]

**Review by tatianahutii — APPROVED** [review]

**Pull request merged**